frontend/lib/views/login.dart

```diff
@@ -19,128 +19,125 @@ class LoginState extends State<Login> {
-                // validator: (value) {
-                //   if (value == null || value.isEmpty) {
-                //     return 'Please a valid UofT email address';
-                //   }
-                //   return null;
-                // },
-              ),
-              SizedBox(height: 16.0),
-              Align(
-                alignment: Alignment.centerLeft,
-                child: Text(
-                  'Password',
-                  style: TextStyle(
-                    fontSize: 16.0,
-                    fontWeight: FontWeight.w700,
+    final logoText = StyleText(
+      text: 'UTMarketplace',
+      style: ThemeText.header.copyWith(
+        color: Theme.of(context).colorScheme.primary,
+      ),
+    );
+
+    final emailField = TextFormField(
+      decoration: InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Enter your UofT email',
+      ),
+      // TODO: REGEX for valid UofT domain email, uncomment when ready
+      // validator: (value) {
+      //   if (value == null || value.isEmpty) {
+      //     return 'Please a valid UofT email address';
+      //   }
+      //   return null;
+      // },
+    );
+
+    final passwordField = TextFormField(
+      obscureText: _obscureText,
+      decoration: InputDecoration(
+        border: OutlineInputBorder(),
+        labelText: 'Enter your password',
+        suffixIcon: IconButton(
+          icon: Icon(
+            _obscureText ? Icons.visibility_off : Icons.visibility,
+          ),
+          onPressed: () {
+            setState(() {
+              _obscureText = !_obscureText;
+            });
+          },
+        ),
+      ),
+      // TODO: Implement email validation and password validation (backend), uncomment when ready
+      // validator: (value) {
+      //   if (value == null || value.isEmpty) {
+      //     return 'Password or email is incorrect';
+      //   }
+      //   return null;
+      // },
+    );
+
+    final loginButton = ElevatedButton(
+      onPressed: () {
+        if (_formKey.currentState!.validate()) {
+          // TODO: Implement login functionality (backend)
+          // If this condition passed, a redirect call to the home
+          // page should be made
+          context.go('/temp_view');
+        }
+      },
+      style: ElevatedButton.styleFrom(
+        minimumSize: Size(double.infinity, 50),
+      ),
+      child: Text('Log In'),
+    );
+
+    final signUpRow = Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: <Widget>[
+        Text('Don\'t have an account?'),
+        TextButton(
+          onPressed: () {
+            // TODO: Implement redirect to sign up page
+          },
+          child: Text('Sign Up'),
+        ),
+      ],
+    );
+
+    final forgotPasswordButton = TextButton(
+      onPressed: () {
+        // TODO: Implement forgot password functionality
+      },
+      child: Text('Forgot Password?'),
+    );
+
+    return SafeArea(
```

Comment:
Nicely done refactor, this will help us quickly identify specific sub-widgets and consider their layout separately as our pages and widgets get more complex.